pyuvdata/telescopes.py

```diff
@@ -94,212 +149,426 @@ def __init__(self):
+                                telescope_ant_inds, :
+                            ]
+                        if "antenna_names" in ant_params_missing:
+                            self.antenna_names = ant_info["antenna_names"][
+                                telescope_ant_inds
+                            ]
+
+                        if "antenna_numbers" in ant_params_missing:
+                            self.antenna_numbers = ant_info["antenna_numbers"][
+                                telescope_ant_inds
+                            ]
+
+            if "antenna_diameters" in telescope_dict.keys() and (
+                overwrite or self.antenna_diameters is None
+            ):
+                antenna_diameters = np.atleast_1d(telescope_dict["antenna_diameters"])
+                if antenna_diameters.size == 1:
+                    known_telescope_list.append("antenna_diameters")
+                    self.antenna_diameters = (
+                        np.zeros(self.Nants, dtype=float) + antenna_diameters[0]
+                    )
+                elif antenna_diameters.size == self.Nants:
+                    known_telescope_list.append("antenna_diameters")
+                    self.antenna_diameters = antenna_diameters
+                else:
+                    if warn:
+                        warnings.warn(
+                            "antenna_diameters are not set because the number "
+                            "of antenna_diameters on known_telescopes_dict is "
+                            "more than one and does not match Nants for "
+                            f"telescope {self.name}."
+                        )
+
+            if "x_orientation" in telescope_dict.keys() and (
+                overwrite or self.x_orientation is None
+            ):
+                known_telescope_list.append("x_orientation")
+                self.x_orientation = telescope_dict["x_orientation"]
+
+        full_list = astropy_sites_list + known_telescope_list
+        if warn and len(full_list) > 0:
+            warn_str = ", ".join(full_list) + " are not set or are being overwritten. "
+            specific_str = []
+            if len(astropy_sites_list) > 0:
+                specific_str.append(
+                    ", ".join(astropy_sites_list) + " are set using values from "
+                    f"astropy sites for {self.name}."
+                )
+            if len(known_telescope_list) > 0:
+                specific_str.append(
+                    ", ".join(known_telescope_list) + " are set using values "
+                    f"from known telescopes for {self.name}."
+                )
+            warn_str += " ".join(specific_str)
+            warnings.warn(warn_str)
 
-    return obj
+        if run_check:
+            self.check(
+                check_extra=check_extra, run_check_acceptability=run_check_acceptability
+            )
+
+    @classmethod
+    def get_telescope_from_known_telescopes(
```

Comment:
I think this should just be `from_known_telescopes`.... this is the usual syntax for object-instantiation class methods.

Also, presumably you're passing in the `known_telescope_dict` so that someone could create their own telescope dict to pass in here? But I would argue that a better interface would be to either:

1. Just let the user update `KNOWN_TELESCOPES` (either by some formal registration of new objects, or simply by doing `KNOWN_TELESCOPES.update(myscope={myscope_dict})`
2. Having a separate classmethod `.from_telescope_params()` (or similar) to which they can just pass the keys that would have been in their entry into `KNOWN_TELESCOPES`... this would be a thin layer over the `from_known_telescopes()` classmethod.

Reply:
The known telescopes passing is basically just for tests. I can try to hide it more.

Reply:
I just updated the docstring to include all the parameters and to make it clear that should only be used for testing. Let me know if I should do more on this front.